virtru-audit-client/auditClient.py

```diff
@@ -0,0 +1,61 @@
+import random
+import hashlib
+import base64
+import requests
+import jwt
+import time
+
+
+class AuditClient:
+    def __init__(self, apiTokenSecret, apiTokenId, apiHost, apiPath):
+        self.apiTokenSecret = apiTokenSecret
+        self.apiTokenId = apiTokenId
+        self.apiHost = apiHost
+        self.apiPath = apiPath
+
+    def fetchRecords(self, req):
+        vjwtString = self.__generateVjwtString(req)
+
+        headers = {
+            'Authorization': 'VJWTv1.0.0 ' + vjwtString.decode()
+        }
+
+        apiUrl = self.apiHost + self.apiPath
+        response = requests.get("https://" + apiUrl,
+                                params=req['query'], headers=headers)
+        return response.json()
+
+    def __generateVjwtString(self, req):
+        tokenSecret = base64.b64decode(self.apiTokenSecret)
+        method = req['method']
+
+        queryParams = self.__generateQueryParams(req['query'])
+        nonce = self.__generateNonce()
+
+        payload = {
+            'sub': self.apiTokenId,
+            'iat': int(time.time()),
+            'jti': nonce,
+            'rsha': self.__generateRsha(method, self.apiHost, self.apiPath, queryParams),
+            'rqps': ','.join(req['query'].keys()),
+            'exp': int(time.time()+120.0)
```

Comment:
What is `120.0`? And why do we specify a float/double if we are going to cast to `int` anyways?

Reply:
120 is the number of seconds. It needs to be a float because time.time() returns a float. I guess we could also have done `int(time.time())+120`

Reply:
I think we should define these explicit numbers with a constant/variable so we know what it means, (e.g. TWO_MINUTES = 120). There also appears to be built-ins for handling date/time arithmetic but I think it's overkill for this.

Reply:
I think Python number coercion works much like it does in JS or Java -- you can add an `int` to a `float` and it will return a `float`.

See my REPL output:
```
$ python
Python 2.7.15 (default, May  1 2018, 16:44:08)
[GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> now = time.time();
>>> now
1536354886.720376
>>> now = time.time()
>>> int(now + 120)
1536355012
>>> int(now + 120.0)
1536355012
```

Reply:
+1 on no magic numbers/strings